### DISCUSSION TITLE
Access the visible model from actions

### DISCUSSION THREAD

**Discussion by iainbeeston** [avo-hq/avo#1404] [Feature requests]
I'd like to be able to access the model that is currently being viewed from actions, because I have many actions that act on the resource that you've editing/viewing.

If I am on a show or edit page, I can get access to a resource for that page but the model is nil (I'm not sure why). At the moment I can access the current model from the `:models` parameter but if the user has selected any associated models (eg. in a has many) then `:models` is set to those models (rather than the currently viewed resource). Perhaps there is already a way to do this and I've missed it?

**Comment by adrianthedev** [comment]
> I'd like to be able to access the model that is currently being viewed from actions, because I have many actions that act on the resource that you've editing/viewing.

You should be able to access the `resource` and `model` in an action if you are on the `show` or `edit` views. If that's not happening, please make a reproduction repo (or use the `dummy` or [avodemo](https://github.com/avo-hq/main.avodemo.com) apps) so we can look into it.
The only times when you don't have access to them is on the index view, or on standalone actions.

> At the moment I can access the current model from the :models parameter but if the user has selected any associated models (eg. in a has many) then :models is set to those models (rather than the currently viewed resource)

Let me make sure I understand correctly. So you are on `Post 1` and have a `has_many` Comments association. When selecting multiple comments, when you run an action, you get all those comments in the `models` argument, but not the parent `Post 1`. Is that correct?
